src/polyglot-notebooks-vscode-common/src/utilities.ts

```diff
@@ -232,6 +232,15 @@ export function parse(text: string): any {
             return Uint8Array.from(buffer.values());
         }
 
+        //handling NaN, Infinity and -Infinity
+        if (value === "Infinity") {
+            return Infinity;
+        } else if (value === "-Infinity") {
+            return -Infinity;
+        } else if (value === "NaN") {
```

Comment:
Micro-optimization micro-nitpick: `NaN` will be by far the most common case here and might should be checked before the others.

Reply:
indeed

Reply:
done